src/main/java/com/notably/model/block/BlockImpl.java

```diff
@@ -0,0 +1,129 @@
+package com.notably.model.block;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Represents a Block in the Notably data structure.
+ */
+public class BlockImpl implements Block {
+    private Block parent;
+    private Title title;
+    private Body body;
+    private List<Block> children;
+
+
+    /**
+     * Initializes a root block i.e. the block has no parent.
+     */
+    public BlockImpl() {
```

Comment:
Actually, thinking about this right now, I feel like perhaps a clearer way to express this is to provide a static method called `getRootBlock()`/`createRootBlock()`? This static method would then return a new block with the `"Root"` title and a `null` parent.

Reply:
Done!